### ISSUE TITLE
PHP Fatal error:  Cannot use Nette\Object as Object because 'Object' is a special class name 

### ISSUE THREAD

**Issue by ghost** [jkuchar/FileDownloader#23]
PHP 7.2.3 (cli) (built: Mar  2 2018 15:48:10) ( NTS )
Copyright (c) 1997-2018 The PHP Group
Zend Engine v3.2.0, Copyright (c) 1998-2018 Zend Technologies
    with Zend OPcache v7.2.3, Copyright (c) 1999-2018, by Zend Technologies
    with Xdebug v2.6.0alpha1, Copyright (c) 2002-2017, by Derick Rethans

the pic is the discription for what is new in PHP 7.2.3
![image](https://user-images.githubusercontent.com/9205255/41074691-d7eacbb4-6a3b-11e8-85f8-4d622cdb7471.png)

this library cannot normal work an php >7 version

**Comment by jkuchar** [comment]
Yep, this is because it uses Nette\Object, which is not supported on PHP7 because of its name. Could you please provide pull request which uses SmartObject instead? https://doc.nette.org/cs/2.4/smartobject

**Comment by jkuchar** [comment]
There should be no more incompatibilities, just this formal one.
